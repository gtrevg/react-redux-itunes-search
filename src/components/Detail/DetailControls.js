import React from 'react';

const DetailControls = props => {
	const { preview } = props;
	if (!preview) return null;
	return (
		<div>
			<audio src={preview} controls>
				Sorry, your browser does not support the audio element.
			</audio>
		</div>
	);
};

export default DetailControls;
