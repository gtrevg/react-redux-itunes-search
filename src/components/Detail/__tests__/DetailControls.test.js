import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import DetailControls from '../DetailControls';

describe('DetailControls component', () => {
	const component = (
		<DetailControls preview="https://audio-ssl.itunes.apple.com/apple-assets-us-std-000001/AudioPreview122/v4/c6/50/11/c6501132-e865-3711-175a-ddb79114e42f/mzaf_3806132797788612279.plus.aac.p.m4a" />
	);

	it('Renders without crashing', () => {
		const div = document.createElement('div');
		ReactDOM.render(component, div);
		ReactDOM.unmountComponentAtNode(div);
	});

	it('Snapshot matchs', () => {
		const tree = renderer.create(component).toJSON();
		expect(tree).toMatchSnapshot();
	});
});
