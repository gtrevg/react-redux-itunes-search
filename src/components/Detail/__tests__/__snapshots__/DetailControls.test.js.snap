// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailControls component Snapshot matchs 1`] = `
<div>
  <audio
    autoPlay={true}
    controls={true}
    src="https://audio-ssl.itunes.apple.com/apple-assets-us-std-000001/AudioPreview122/v4/c6/50/11/c6501132-e865-3711-175a-ddb79114e42f/mzaf_3806132797788612279.plus.aac.p.m4a"
  >
    Sorry, your browser does not support the audio element.
  </audio>
</div>
`;
